salesforce/modules/rest/client.bal

```diff
@@ -0,0 +1,347 @@
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, id]);
+        return check self.salesforceClient->delete(path);
+    }
+
+    # Executes the specified SOQL query.
+    #
+    # + soql - SOQL query
+    # + returnType - The payload, which is expected to be returned after data binding.
+    # + return - `stream<{returnType}, error?>` if successful. Else, the occurred `error`.
+    @display {label: "Get Query Result"}
+    isolated remote function query(@display {label: "SOQL Query"} string soql, typedesc<record {}> returnType = <>)
+                                    returns @display {label: "SOQL Result"} stream<returnType, error?>|error = @java:Method {
+        'class: "io.ballerinax.salesforce.ReadOperationExecutor",
+        name: "getQueryResult"
+    } external;
+
+    private isolated function processGetQueryResult(typedesc<record {}> returnType, string receivedQuery)
+                                                    returns stream<record {}, error?>|error {
+        string path = utils:prepareQueryUrl([API_BASE_PATH, QUERY], [Q], [receivedQuery]);
+        SOQLQueryResultStream objectInstance = check new (self.salesforceClient, path);
+        stream<record {}, error?> finalStream = new (objectInstance);
+        return finalStream;
+    }
+
+    # Executes the specified SOSL search.
+    #
+    # + sosl - SOSL search query
+    # + returnType - The payload, which is expected to be returned after data binding.
+    # + return - `stream<{returnType}, error?>` record if successful. Else, the occurred `error`.
+    @display {label: "SOSL Search"}
+    isolated remote function search(@display {label: "SOSL Search Query"} string sosl, typedesc<record {}> returnType = <>)
+                                    returns @display {label: "SOSL Result"} stream<returnType, error?>|error = @java:Method {
+        'class: "io.ballerinax.salesforce.ReadOperationExecutor",
+        name: "searchSOSLString"
+    } external;
+
+    private isolated function processSearchSOSLString(typedesc<record {}> returnType, string searchString)
+                                                    returns stream<record {}, error?>|error {
+        string path = utils:prepareQueryUrl([API_BASE_PATH, SEARCH], [Q], [searchString]);
+        SOSLSearchResult objectInstance = check new (self.salesforceClient, path);
+        stream<record {}, error?> finalStream = new (objectInstance);
+        return finalStream;
+    }
+
+    //Describe SObjects
+    # Lists the available objects and their metadata for your organization and available to the logged-in user.
+    #
+    # + return - `OrgMetadata` record if successful or else `sfdc:Error`
+    @display {label: "Get Available Objects"}
+    isolated remote function describeAvailableObjects() 
+                                                      returns @display {label: "Organization Metadata"} 
+                                                      OrgMetadata|error {
```

Comment:
Formatting issue

Reply:
Fixed in 8bfb122ccf270a0868c40a7ea6ddaef1d3a79a39